packages/web/components/templates/article/ArticleContainer.tsx

```diff
@@ -36,6 +37,11 @@ export function ArticleContainer(props: ArticleContainerProps): JSX.Element {
   const [showShareModal, setShowShareModal] = useState(false)
   const [showReportIssuesModal, setShowReportIssuesModal] = useState(false)
   const [fontSize, setFontSize] = useState(props.fontSize ?? 20)
+  const highlightHref = useRef(window.location.hash ? window.location.hash.split('#')[1] : null)
```

Comment:
How come this uses `useRef`? Is it so we don't access window while server side rendering?

Reply:
We are not expecting to perform any changes after the initial Href of the url on load. That's why I am using useRef, I was not thinking about serverside since the function to scroll to the position runs in a useEffect

Reply:
[@jacksonh](https://github.com/jacksonh) Yeah, the highlights are being added asynchronously to the page and we need to know that they are ready before running the function to scroll to either the last read position or highlight with an id. We could use prop drilling to also achieve this. Another way we can go about this is if we take note of the position of the highlight parent and immediately scroll rather than using the highlight id

Reply:
We are not expecting to perform any changes after the initial Href of the URL on load. That's why we are using useRef, we weren't not thinking about server side since the function to scroll to the position runs in a useEffect.